src/pybind/mgr/dashboard/frontend/src/app/shared/directives/stateful-tab.directive.spec.ts

```diff
@@ -0,0 +1,8 @@
+import { StatefulTabDirective } from './stateful-tab.directive';
+
+describe('StatefulTabDirective', () => {
+  it('should create an instance', () => {
+    const directive = new StatefulTabDirective(null);
+    expect(directive).toBeTruthy();
```

Comment:
Could you please add a more meaningful test? A good test would be to make sure that the component checks the local storage and set's it on event change.